teps/0091-trusted-resources.md

```diff
@@ -407,6 +401,71 @@ How does this proposal affect the reusability, simplicity, flexibility
 and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
 -->
 
+### Reusability
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#reusability
+
+- Are there existing features related to the proposed features? Were the existing features reused?
+- Is the problem being solved an authoring-time or runtime-concern? Is the proposed feature at the appropriate level
+authoring or runtime?
+-->
+
+There are no existing features in Tekton which can be reused.
+
+This problem is runtime-concern and the proposed feature is at runtime level.
+
+
+### Simplicity
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#simplicity
+
+- How does this proposal affect the user experience?
+- What’s the current user experience without the feature and how challenging is it?
+- What will be the user experience with the feature? How would it have changed?
+- Does this proposal contain the bare minimum change needed to solve for the use cases?
+- Are there any implicit behaviors in the proposal? Would users expect these implicit behaviors or would they be
+surprising? Are there security implications for these implicit behaviors?
+-->
+
+This proposal doesn't have any effect if users don't enable it. If enabled, users need to sign the resources and config the public key in cluster to bypass verification, the mutating webhook will be skipped for API resources to avoid the failing verification.
```

Comment:
nit: Shouldn't the webhook be removed in either case? It would be weird if different requests were produced depending on whether this is enabled, since this is more around enforcement.

Reply:
My guess is this will probably be wanted sooner rather than later, since it's much easier for maintenance to have both paths have the same logic rather than have to have different assumptions depending on whether the mutating webhook ran or not.

To support verification you're already going to make sure similar defaulting behavior happens in the reconciler anyway, and there shouldn't be a harm in doing this post-mutating webhook for non-verified objects since they should produce the same result. If defaulting is happening at reconcile for all requests and we're allowing certain requests (i.e. verified resources) to skip the mutating webhook, then the mutating webhook isn't adding value and can just be removed since even non-verified resources should be able to safely skip the webhook as well.

This doesn't change this design, just more-so an FYI to other projects that this will be happening since this will change the stored API object for created objects which may be considered a backwards incompatible change - though I suspect most other tools are agnostic to the defaulting that the webhook is doing and won't even notice it's gone. If we need to follow the deprecation policy strictly here I'd rather do it sooner than later, since the main reason I can see keeping the webhook around is for backwards-compatibility.

Reply:
Ok got it!